### PR TITLE
Implement Redux

### PR DESCRIPTION
**Pull request by thongquach-eh**
- Implement Redux using hooks
- Move types to a separate file

### CODE REVIEW

File: types.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// @flow
+export type RootStackParamList = {
```
Comment (ttkien) [inline types.js]: I cannot find any place using this type.

File: components/UserListScreen.js [hunk 2]
```diff
@@ -2,86 +2,63 @@
...
-    large: ?string,
-    medium: ?string,
-    thumbnail: ?string,
-  },
-};
-
-const renderUser = (user: UserProps): React.Node => (
-  <TouchableHighlight style={styles.item}>
-    <>
+const renderUser = (user: User, navigation): React.Node => (
```
Comment (ttkien) [inline components/UserListScreen.js]: Should create a component instead of function

File: types.js (hunk 1 above)
Comment (thongquach-eh) [reply types.js]: I have removed this type, it will be used in my upcoming PR.

File: components/UserListScreen.js (hunk 2 above)
Comment (thongquach-eh) [reply components/UserListScreen.js]: I've moved this to a separate component

**Review by ttkien — APPROVED** [review]

**Pull request merged**